src/gql/GQLWrapper.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useState } from "react";
+import { ApolloClient } from "apollo-client";
+import { ApolloProvider } from "@apollo/react-hooks";
+import { HttpLink } from "apollo-link-http";
+import { InMemoryCache, NormalizedCacheObject } from "apollo-cache-inmemory";
+import {
+  addMockFunctionsToSchema,
+  introspectSchema,
+  makeExecutableSchema
+} from "graphql-tools";
+import { printSchema } from "graphql/utilities/schemaPrinter";
+import { SchemaLink } from "apollo-link-schema";
+
+interface ClientLinkParams {
+  credentials?: string;
+  gqlURL?: string;
+  isDevelopment?: boolean;
+  isTest?: boolean;
+  schemaString?: string;
+}
+
+export const getClientLink = async ({
+  gqlURL,
+  isDevelopment,
+  isTest,
+  schemaString,
+  credentials
+}: ClientLinkParams): Promise<HttpLink | SchemaLink> => {
+  const httpLink = new HttpLink({
+    uri: gqlURL,
+    credentials
+  });
+
+  if (isDevelopment || isTest) {
+    try {
+      const executableSchema = makeExecutableSchema({
+        typeDefs: schemaString
+          ? schemaString
+          : printSchema(await introspectSchema(httpLink))
+      });
+      addMockFunctionsToSchema({ schema: executableSchema });
+      return new SchemaLink({ schema: executableSchema });
+    } catch (e) {
+      console.warn(
+        "Unable to initiate mock server. If this was unintended, provide a valid value for the REACT_APP_GQL_URL or REACT_APP_SCHEMA_STRING environment variables."
+      );
+      return new HttpLink();
+    }
+  }
+  return httpLink;
+};
+
+const cache = new InMemoryCache();
```

Comment:
It seems weird to have this outside the scope of a function. I would move it into getGQLClient

Reply:
it's not weird. having it outside the scope of the function is technically more performant because it is only ever invoked once rather than on re-render

Reply:
Do we want the same cache for each GQL client?

Reply:
there's only one gql client

Reply:
the useEffect callback will only be called once for the lifetime of GraphQLWrapper. If for whatever reason GraphQLWrapper unmounts and remounts, it will use the same cache from before it unmounted. For now we don't expect GraphQLWrapper to unmount and remount for the lifetime of App.

Reply:
If that's the desired behavior then what you have right now is correct (I realize it's mostly a moot point anyway)